### PR TITLE
Feature/implementation deck of cards to frond

### PR DESCRIPTION
**Pull request by Dakova1994**

### CODE REVIEW

File: Projekt/src/app/app.component.html [hunk 1]
```diff
@@ -18,3 +18,13 @@ <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular bl
...
 </ul>
 
+<h2>Here is your cards</h2>
+<div *ngIf="deck && pile">
+  <div *ngFor="let card of pile.cards">
+    <img src="{{card.image}}">
+  </div>
+</div>
+<div>
+  <input type="number" (input)="cardsNumber = $event.target.value" min="1" max="52">
```
Comment (emosGambler) [inline Projekt/src/app/app.component.html]: I think we could do something like this: `max="deck.remaining"`. Try if it works :)

File: Projekt/src/app/app.component.html [hunk 2]
```diff
@@ -18,3 +18,13 @@ <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular bl
...
+</div>
+<button (click)="onClickMe()">Click me!</button>
```
Comment (emosGambler) [inline Projekt/src/app/app.component.html]: Change this method name into `drawCards()`

Comment (emosGambler) [inline Projekt/src/app/app.component.html]: Change `Click me!` to `Draw cards`

File: Projekt/src/app/app.component.ts [hunk 3]
```diff
@@ -1,25 +1,30 @@
 import { Component, OnInit } from '@angular/core';
 import { CardsService} from './app.service';
-import { IDeck } from './deck.model';
+import { IDeck, IPile } from './deck.model';
+import { NgModel } from '@angular/forms';
```
Comment (emosGambler) [inline Projekt/src/app/app.component.ts]: Unused import

File: Projekt/src/app/app.component.ts [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
-		this.cardsService.getShuffledDeck().subscribe( shuffledDeck  => {
-			this.deck = shuffledDeck ;
-			console.log('moje rezultaty:', this.deck);
-			this.cardsService.getCards(this.deck.deck_id, this.deck.remaining).subscribe( newCards  =>{
-				this.twoCards = newCards ;
-				console.log('dwie karty:', this.twoCards);
-			});
-		});
-	}
+    title = 'app';
```
Comment (emosGambler) [inline Projekt/src/app/app.component.ts]: make all of these variables `public`

File: Projekt/src/app/app.component.ts [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
+    constructor(private cardsService: CardsService) {}
+    ngOnInit() {
+        this.cardsService.getShuffledDeck().subscribe( shuffledDeck  => {
+            this.deck = shuffledDeck ;
+        });
+    }
+    onClickMe(){
+        this.cardsService.getCards(this.deck.deck_id, this.cardsNumber).subscribe( pile  => {
+            this.pile = pile ;
+            console.log("pile: ", this.pile)
```
Comment (emosGambler) [inline Projekt/src/app/app.component.ts]: remove these console.logs

File: Projekt/src/app/app.service.ts [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
 
 @Injectable()
-export class CardsService{
-    constructor(private http: HttpClient){}
+export class CardsService {
+    constructor(private http: HttpClient ) {}
 
-    public getShuffledDeck(){
-        return this.http.get<IDeck>("https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1", )
+    public getShuffledDeck() {
```
Comment (emosGambler) [inline Projekt/src/app/app.service.ts]: `public getShuffledDeck(numberOfDecks: number = 1) {`

File: Projekt/src/app/app.service.ts [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
+        return this.http.get<IDeck>('https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1');
```
Comment (emosGambler) [inline Projekt/src/app/app.service.ts]: `${numberOfDecks}`

File: Projekt/src/app/app.service.ts [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
 
 @Injectable()
-export class CardsService{
-    constructor(private http: HttpClient){}
+export class CardsService {
+    constructor(private http: HttpClient ) {}
 
-    public getShuffledDeck(){
-        return this.http.get<IDeck>("https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1", )
+    public getShuffledDeck(numberOfDecks: number = 0) {
```
Comment (emosGambler) [inline Projekt/src/app/app.service.ts]: `numberOfDecks` should equals 1.

File: Projekt/src/app/app.service.ts [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
+        return this.http.get<IDeck>('https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1');
```
Comment (emosGambler) [inline Projekt/src/app/app.service.ts]: `return this.http.get<IDeck>(`https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=${numberOfCards}`);

Comment (Dakova1994) [reply Projekt/src/app/app.service.ts]: fixed

**Pull request merged**